### DISCUSSION TITLE
Number of issues in new Gitbook publish site

### DISCUSSION THREAD

**Discussion by ahcsco** [GitbookIO/community#507] [2. Published content] [answered]
Experiencing a number of issues in the new Gitbook publish site.
Overall, definitely a step in the right direction, but issues need to be addressed. 

**Performance related:**
-Pages are generally taking longer to load. Experiencing laggy loads, even up to 10 seconds.
-Updates/changes done to spaces are not reflected instantly (even after manual page refresh). Taking several hours to reflect.
-Auto page refresh upon new changes/merge is also not working anymore.
-Same for changes to site customization settings. Also not getting reflected instantly. Tested with font change, no update on published site after several hours.
-Some obvious improvements in PDF export but the export is taking too long to render or not loading at all (page breaks down with error) and when it does, it does with “Loading...” text in the pdf file. Plus, in dark mode, the font colors in pdf are also all off to very light grey.
-Editor issue: Occasionally the Merge button does not work and freezes with the merge never finishing. Need to refresh the page to get the merge button back and try again.
-Content sometimes randomly going back to older commits on the publish site. Need to refresh to get the latest. Could be related to above issue with performance. Need to be checked.

**Usability related:**
-Search is not categorizing pages by page groups anymore. The old search did it way better.
-The editor styling should be the same as the publish site styling. Now it’s all guesswork (e.g. with headings, bullets, notes, code, tables, formatting, etc) how it’ll show up because both use different styling sheets.
-Old site used to automatically add the text “following articles in this section” to empty pages. Not anymore.  Link boxes appear fine though.
-External links are opening in the same tab instead of a new tab. 

Good to look into:
-Place the dark/light mode toggle switch in the right panel. I find it very strange to have to scroll all the way down to to footer to find dark/light mode switch. Odd UX on long pages.
-Strange gridlines appear during page loads. Probably should have the usual load animation.

**Comment by alemscoder** [comment]
I would like to add that the style for the Info hint don't really look like a info hint

**Reply by Zoce** [reply]
hace 4 dias no actualiza los cambios

**Comment by LittleWormOnAHook** [comment]
I've been having issues for days, updated content merges with no errors, but url does not update..Help

**Reply by Zoce** [reply]
yo igual!

**Comment by SamyPesse** [comment, marked as answer]
Thank you for the complete feedback, it's greatly appreciated.

> Pages are generally taking longer to load. Experiencing laggy loads, even up to 10 seconds.

Can you share specific URLs? it'll help us debug the issue. Overall you should see fast performances on all pages, especially after the content has been loaded once by a visitor.

> Updates/changes done to spaces are not reflected instantly (even after manual page refresh). Taking several hours to reflect.
> Same for changes to site customization settings. Also not getting reflected instantly. Tested with font change, no update on published site after several hours.

We are working on this and currently deploying a change that should improve our data cache invalidation to see changes being live after max 5min.

> Auto page refresh upon new changes/merge is also not working anymore.

We are not sure we'll keep supporting this one. Our new architecture is designed to optimize performance and render the content as much as possible on the server. At the moment, getting the latest change requires refreshing the page.

We'll keep you posted as we think more about this one.

> Some obvious improvements in PDF export but the export is taking too long to render or not loading at all (page breaks down with error) and when it does, it does with “Loading...” text in the pdf file. Plus, in dark mode, the font colors in pdf are also all off to very light grey.

If the space is public, can you share its URL? we'll investigate the issue.

> Editor issue: Occasionally the Merge button does not work and freezes with the merge never finishing. Need to refresh the page to get the merge button back and try again.

It's not related to this release. I passed it down to our team.

> Content sometimes randomly going back to older commits on the publish site. Need to refresh to get the latest. Could be related to above issue with performance. Need to be checked.

Yes, it's definitely related; it should soon be improved.

> Search is not categorizing pages by page groups anymore. The old search did it way better.

Can you tell me more about this one? maybe by sharing an example of you'd expect.

> The editor styling should be the same as the publish site styling. Now it’s all guesswork (e.g. with headings, bullets, notes, code, tables, formatting, etc) how it’ll show up because both use different styling sheets.

We are working on better aligning the 2, but there were already a bunch of differences, and there will still be differences.

> Old site used to automatically add the text “following articles in this section” to empty pages. Not anymore. Link boxes appear fine though.

The reason why we removed it is because this text was making assumptions on the type of content; a lot of our users use GitBook for stuff that are not "articles"; we wanted to be more generic.

> External links are opening in the same tab instead of a new tab.

We'll look into it.

> Place the dark/light mode toggle switch in the right panel. I find it very strange to have to scroll all the way down to to footer to find dark/light mode switch. Odd UX on long pages.

This is a design decision; we believe that most readers actually do not want to change light/dark mode since we now default to the "system" one. So instead of putting this in the header and distracting the readers from the content, we pushed it to the footer.

> Strange gridlines appear during page loads. Probably should have the usual load animation.

Again this is a design decision, by having a loading placeholder that mimic the content (some blocks could be improved), we limit the perception of content shift.

**Reply by ahcsco** [reply]
Thanks for the detailed reply and for passing over the issues to your team.

I couldn't say I've experienced a better performance so far. It's either the same or a tad worse (esp on longer pages), but maybe that's just me.

What we are really looking forward to is changes (including site settings) getting reflected within a couple of minutes if not seconds (+ the merge button freeze). I understand from your reply some improvements might still be on the way. So fingers crossed and hopefully, we'll see the performance getting some more love. We have recently purchased Gitbook and are in the middle of evaluating it thoroughly. Being one of the biggest IT companies in the world, it's a great deal for us. So far it's made a pretty good case and I hope it continues to do so. Our site is still undergoing testing so although it's publicly visible, it's still not public. I would be more than happy to share the link with you/team privately for investigation. 

Some of the issues really pop up randomly, so it becomes difficult to reproduce them but perhaps the ones like PDF dark mode can simply be reproduced by toggling to dark mode and checking the colors and output. With the search, my point was that the old site did a handy categorization of pages using page groups. That gives a good context to users about what they are trying to find and where. The new search lists out pages without that specific context, for example, an "Administration" page group result, and a "Usage" page group result that are similar but can be differentiated based on context. 

That being said, the new changes do definitely provide a better appearance experience.

**Reply by ahcsco** [reply]
At least the merge freeze, I was able to capture. To guess, it seems some kind of backend timeout that causes this behavior (disables the button without finishing the merge; need to refresh the page to get the button back and there's a risk your changes may not be there after refresh).

Going on for several minutes, captured 30 seconds or so. 

https://github.com/GitbookIO/community/assets/112613597/d70bcdf3-6724-4e53-8064-f269790dc64b

**Comment by rachel-shapira** [comment]
Any update on fixing external links so that they open in a new tab?

**Comment by vietman78** [comment]
I think Gitbooks may have used AI for these updates. They are terrible.

**Comment by ahcsco** [comment]
Is there any update on the remaining issues?

I raised it earlier probably. The content in the editor does not occasionally update to the new commit after pressing merge. You need to click edit again and re-merge to have it show up. Very frustrating and likely to cause a lot of update issues with many people editing (no one will ever guess that the app will fail to update their change). 

Also, the merge button freeze is still not resolved. I keep seeing it all the time. Need to refresh the page every time before starting changes. 

These are bread-and-butter issues for a CMS and this haphazard update is still hurting us. Please have it looked into on prio. It's getting in the way of work. Thanks.

**Comment by benitav** [comment]
> Pages are generally taking longer to load. Experiencing laggy loads, even up to 10 seconds

We are seeing the same. We're also finding that when visiting a page directly (e.g. https://docs.powersync.com/client-sdk-references/react-native-and-expo) it loads relatively instantly, but when selecting it from the sidebar it takes 3-5 seconds to load and displays gridlines during load.
